tensorflow/core/util/rocm_solvers.h

```diff
@@ -78,8 +88,62 @@ class ROCmSolver {
       gtl::ArraySlice<int> candidate_input_indices, DataType type,
       const TensorShape& shape, Tensor* input_alias_or_new_scoped_tensor);
 
+  static void CheckLapackInfoAndDeleteSolverAsync(
+    std::unique_ptr<ROCmSolver> solver, 
+    const std::vector<DeviceLapackInfo>& dev_lapack_info, 
+    std::function<void(const Status&, const std::vector<HostLapackInfo>&)>
+    info_checker_callback);  
+
+  static void CheckLapackInfoAndDeleteSolverAsync(
+    std::unique_ptr<ROCmSolver> solver,
+    const std::vector<DeviceLapackInfo>& dev_lapack_info,
+    AsyncOpKernel::DoneCallback done); 
+
   OpKernelContext* context() { return context_; }
 
+  template <typename Scalar>
+  ScratchSpace<Scalar> GetScratchSpace(const TensorShape& shape,
+                                       const std::string& debug_info,
+                                       bool on_host);
+  template <typename Scalar>
+  ScratchSpace<Scalar> GetScratchSpace(int64 size,
+                                       const std::string& debug_info,
+                                       bool on_host);
+  // Returns a DeviceLapackInfo that will live for the duration of the
+  // ROCmSolver object.
+  inline DeviceLapackInfo GetDeviceLapackInfo(int64 size,
+                                              const std::string& debug_info);
+
+
+  // ====================================================================
+  // Wrappers for ROCSolver start here
+  //
+  // The method names below
+  // map to those in ROCSolver, which follow the naming
+  // convention in LAPACK see
+
+  // LU factorization.
+  // Computes LU factorization with partial pivoting P * A = L * U.
+  template <typename Scalar>
+  Status getrf(int m, int n, Scalar* dev_A, int lda, int* dev_pivots, int* info);
```

Comment:
Can we try to keep the API name + interface, common between `ROCmSolver` / `CudaSolver` classes. 

Ideally we would want to have only one class + header file ( `GpuSolver` +  `gpu_solver.h` ) and platform specific   implementations for them ( in `rocm_solver.cpp` and `cuda_solver.cpp` ). There is a good chance we will be requested to do this too when we try to upstream this change.